fresh install with code-sharing not working

**Environment**
Angular CLI: 9.1.8
Node: 12.16.3
OS: darwin x64

tns@6.7.4

**Describe the bug**
Can't install the dependencies. The module `@angular/http` is required but it does no longer exist for angular 9

```bash
Installing packages...npm WARN deprecated tslint@6.1.2: TSLint has been deprecated in favor of ESLint. Please see https://github.com/palantir/tslint/issues/4534 for more information.
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm ERR! code ETARGET
npm ERR! notarget No matching version found for @angular/http@~9.1.0.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
npm ERR! notarget 
npm ERR! notarget It was specified as a dependency of 'app'
npm ERR! notarget 

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/Daweed/.npm/_logs/2020-06-17T04_08_53_846Z-debug.log
✖ Package install failed, see above.
The Schematic workflow failed. See above.
```

**To Reproduce**
run `ng new --collection=@nativescript/schematics app --shared --style=scss`

**Expected behavior**
Install without error

--- a/package.json
+++ b/package.json
@@ -15,6 +15,12 @@
     "lint": "tslint --project ./tsconfig.json -c ./tslint.json 'src/**/*.ts'",
     "lint:fix": "tslint --project ./tsconfig.json -c ./tslint.json 'src/**/*.ts' --fix"
   },
+  "ng-add": {
+    "save": "devDependencies"
+  },
+  "ng-new": {
+    "save": "devDependencies"
+  },
   "schematics": "./src/collection.json",
   "dependencies": {
     "@angular-devkit/core": "~9.1.0",

--- a/src/add-ns/index.ts
+++ b/src/add-ns/index.ts
@@ -374,7 +374,6 @@ const addDependencies = () => (tree: Tree, context: SchematicContext) => {
 
   const devDepsToAdd = {
     'nativescript-dev-webpack': '~1.5.0',
-    '@nativescript/schematics': '~2.0.0',
     '@nativescript/tslint-rules': '~0.0.5',
   };
   packageJson.devDependencies = {...devDepsToAdd, ...packageJson.devDependencies};

--- a/src/add-ns/index_spec.ts
+++ b/src/add-ns/index_spec.ts
@@ -84,7 +84,6 @@ describe('Add {N} schematic', () => {
             expect(dependencies['reflect-metadata']).toBeDefined();
 
             expect(devDependencies['nativescript-dev-webpack']).toBeDefined();
-            expect(devDependencies['@nativescript/schematics']).toBeDefined();
             expect(devDependencies['@nativescript/tslint-rules']).toBeDefined();
         });
 

--- a/src/ng-new/application/_files/package.json
+++ b/src/ng-new/application/_files/package.json
@@ -11,7 +11,6 @@
     "@angular/compiler": "~9.1.0",
     "@angular/core": "~9.1.0",
     "@angular/forms": "~9.1.0",
-    "@angular/http": "~9.1.0",
     "@angular/platform-browser": "~9.1.0",
     "@angular/platform-browser-dynamic": "~9.1.0",
     "@angular/router": "~9.1.0",

--- a/src/ng-new/application/_files/package.json
+++ b/src/ng-new/application/_files/package.json
@@ -25,8 +25,7 @@
   "devDependencies": {
     "@angular/cli": "~9.1.0",
     "@angular/compiler-cli": "~9.1.0",
-    "@angular-devkit/core": "~9.1.0",
-    "@nativescript/schematics": "~2.0.0",<% if(webpack) { %>
+    "@angular-devkit/core": "~9.1.0",<% if(webpack) { %>
     "nativescript-dev-webpack": "~1.5.0",
     "@ngtools/webpack": "~9.1.0",
     <% } %>"typescript": "~3.8.3"

--- a/src/ng-new/shared/_files/package.json
+++ b/src/ng-new/shared/_files/package.json
@@ -39,7 +39,6 @@
     "@angular/cli": "~9.1.0",
     "@angular/compiler-cli": "~9.1.0",
     "@angular-devkit/build-angular": "~0.901.0",
-    "@nativescript/schematics": "~2.0.0",
     "@nativescript/tslint-rules": "~0.0.5",
     "@types/jasmine": "~3.5.0",
     "@types/jasminewd2": "~2.0.3",

--- a/src/ng-new/shared/_files/package.json
+++ b/src/ng-new/shared/_files/package.json
@@ -23,7 +23,6 @@
     "@angular/compiler": "~9.1.0",
     "@angular/core": "~9.1.0",
     "@angular/forms": "~9.1.0",
-    "@angular/http": "~9.1.0",
     "@angular/platform-browser": "~9.1.0",
     "@angular/platform-browser-dynamic": "~9.1.0",
     "@angular/router": "~9.1.0",

